cli/pcluster/config/param_types.py

```diff
@@ -648,25 +648,89 @@ def to_cfn(self):
         return cfn_params
 
 
+class EC2IAMPolicyInclusionRule(object):
```

Comment:
I think you're right that it's a good idea to move the classes to another module. I moved them to pcluster.config.iam_policy_rules.py. Let me know if you think they should go somewhere else.

The reason I did not make this an abstract base class is because the class is only used via class methods. So even if I were to use that, the "abstract" class method could still be invoked because there is never an attempt to create an instance of the class:
```
import abc

class TestABC(abc.ABC):
    @classmethod
    @abc.abstractmethod
    def someabstractmethod(cls):
        print('This is an abstract method')
```
```
>>> import test_abc
>>> test_abc.TestABC.someabstractmethod()
This is an abstract method
```

In some of the examples I've seen, they prevent this by having the base class's classmethod return an instance of the class:
```
import abc

class TestABC(abc.ABC):
    @classmethod
    @abc.abstractmethod
    def someabstractmethod(cls):
        return cls()
```
```
>>> import test_abc
>>> test_abc.TestABC.someabstractmethod()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/tilne/repos/aws-parallelcluster-cookbook/util/test_abc.py", line 7, in someabstractmethod
    return cls()
TypeError: Can't instantiate abstract class TestABC with abstract methods someabstractmethod
```

I can make the change, but the process for ensuring that classes don't inherit the interface without implementing it is still 'manual'.

Reply:
You right, thank you for the detailed explanation.
I agree we can avoid this "manual" check.